### ISSUE TITLE
GeDoSaTo is BROCKEN

### ISSUE THREAD

**Issue by lextra2** [PeterTh/gedosato#376]
The newest version of GeDoSaTo (0.21.2310) does no longer work on ANY GAME.
As many have already pointed out it has something to do with the "D3DCompiler_47.dll".

Every single game i've tried (Dead Space, Alan Wake, Mirrors Edge) does neither show the resolutions set in GeDoSaTo nor any other supported effects (fx).

Copying the "D3DCompiler_47.dll" to either the games .exe folder, or to the System32/SysWOW64 folder will cause the game to either report an c++ runtime error or the game won't even boot.

I've reinstalled all visual c++ redistributables from 2005 to 2015 with all updates, all .net frameworks, from 2.0 and up, and direcX. I uninstalled and reinstalled the games. Nothing changed.

I even reinstalled windows.

Absolutely nothing has helped. This software is therefore completly broken.

**Comment by LordJBeckman** [comment]
I have read several reports on this, makes me wonder if it could be a OS related issue, GeDoSaTo does work here (Win10, insider, build 14393.5) but D3DCompiler_47.dll is related to DirectX 11.2 from Windows 8.1 I believe though I do not know what it does for GeDoSaTo as it's D3D11 functionality is still not entirely finished.

The C++ compiler issue is interesting too, I keep those up to date via a cumulative bundle but I don't think they would be part of the DirectX files directly so the only thing that comes to mind is the OS thing but that is just a guess on my part, hopefully Durante can fix this once he has time to further work on this project. :)

http://forums.guru3d.com/showpost.php?p=5311169&postcount=143

That's the VC++ installer compilation.

**Comment by lextra2** [comment]
I thought about it being OS related. I'm on windows 7 right now, but might install windows 8.1 if i feel like it.
That installer compilation is quite neat. 
I will give it a try and report back.

edit: Alright. So installing the VC++ installer compilation did remove the runtime error message, but resolutions or post.fx are still not showing up. I will now install windows 8.1 and report back again.

edit2: Installed 8.1. It didn't change anything. It's like GeDoSaTo isn't even there.

**Comment by LordJBeckman** [comment]
For now what you can do until this is properly fixed is that you can use the Github function to pull a earlier version of GeDoSaTo and downgrade to a version that worked for you before these changes were made and hopefully that will do in the meantime. :)

That would be this build before the June 7th changes.
https://github.com/PeterTh/gedosato/tree/b276be5a00ea30a64b841b904aba569d5268669e

Just download that as a .zip archive via the green button to grab this older build of GeDo and thus undo the latest commits. :)

Alternatively there is a commit specifically for d3dcompiler_47.dll here:
https://github.com/PeterTh/gedosato/commit/493ab1b3b41523ce0e09555d401fb614826f5bfb

So you could also try to download that build:
https://github.com/PeterTh/gedosato/tree/493ab1b3b41523ce0e09555d401fb614826f5bfb

And just replace D3DCompiler_47.dll with this older file before a newer commit replaced it with a different one if that could be what is causing this entire issue. :)

https://github.com/PeterTh/gedosato/commit/13e566997a81e1246ba9e57fda1d40dcf8f88db2
(This is the updated file which seems to be causing this entire problem but why exactly that is I can't say.)

Hopefully one of these workarounds will resolve the problem until Durante can do a permanent fix with a future update to GeDoSaTo. :)
(Well downgrading the software to before these changes were introduced should definitively work with little loss of functionality besides some things specific to Stranger of Sword City and probably some code for D3D11 and other render API compatibility code but as D3D9 is still the only one supported this doesn't matter for just using the program.)

**Comment by lextra2** [comment]
Well thank you very much. The February 28. build does work, while the June 7. build gave me the "api-ms-win-core-heap-l2-1-0.dll is missing from your computer" error.

Could you tell me how i navigate between older builds? So incase something like this happens again i can use an older build instead.

**Comment by LordJBeckman** [comment]
Yeah it took me a while before I understood how to do that too, it's been suggested before but I only recently learned how it works myself. :)

When you check the commits to a project - https://github.com/PeterTh/gedosato/commits/master

You can click on any one of them - https://github.com/PeterTh/gedosato/commit/59f1e9706542aa1dcba9e1ad66493afaf7a3b1ba
(This is the latest one.)

And then you just press the "Browse files" button - https://github.com/PeterTh/gedosato/tree/59f1e9706542aa1dcba9e1ad66493afaf7a3b1ba

And it'll take you to this specific commit where you can then download the files from it. :)

**Comment by lextra2** [comment]
Alright, got it, thanks!
